packages/web/src/components/molecules/Console/Console.tsx

```diff
@@ -43,141 +47,186 @@ export interface ConsoleRef {
-  const getTopPx = (line: number) => lineHeight * getTop(line);
-  const getCenterPx = (line: number) => getTopPx(line) + (getSize(line) * lineHeight) / 2;
-
-  const getViewportTop = useLastCallback(() => Math.floor((containerRef.current?.scrollTop || 0) / lineHeight));
-  const getViewportHeight = useLastCallback(() => Math.ceil((containerRef.current?.clientHeight || 0) / lineHeight));
-  const getViewport = () => {
-    const prerender = Math.max(Math.round(getViewportHeight() / 2), 30);
-    const viewportStart = Math.max(getViewportTop() - prerender, 0);
-    const viewportEnd = Math.min(viewportStart + getViewportHeight() + 2 * prerender, total - 1);
-    return {start: viewportStart, end: viewportEnd};
-  };
-  const getViewportLast = () => Math.ceil(Math.min(1 + getViewportTop() + getViewportHeight(), total));
-
-  // Keep information about line width
-  const maxCharactersCount = useMemo(() => processor.getMaxLineLength(), [processor]);
-  const minWidth = wrap ? 0 : baseWidth + characterWidth * maxCharactersCount;
-
-  // Compute current position
-  const {start: viewportStart, end: viewportEnd} = getViewport();
-  const {index: start, start: visualStart} = useMemo(
-    () => getVisualLine(viewportStart + 1),
-    [processor, maxCharacters, viewportStart]
-  );
-  const {index: end, end: visualEnd} = useMemo(
-    () => getVisualLine(viewportEnd + 1),
-    [processor, maxCharacters, viewportEnd]
-  );
-
-  const displayed = useMemo(() => processor.getProcessedLines(start, end + 1), [processor, start, end]);
-
-  const beforeCount = visualStart;
-  const afterCount = total - visualEnd;
-  const beforePx = Math.max(0, Math.floor(beforeCount * lineHeight));
-  const afterPx = Math.max(0, Math.floor(afterCount * lineHeight));
-  const styleTop = useMemo(() => ({height: `${beforePx}px`, width: `${minWidth}px`}), [beforePx, minWidth]);
-  const styleBottom = useMemo(() => ({height: `${afterPx}px`}), [afterPx]);
-
-  const scrollToLine = useLastCallback((line: number) => {
-    containerRef.current?.scrollTo(0, getCenterPx(line) - containerRef.current?.clientHeight / 2);
-  });
-  const getLineRect = useLastCallback((line: number) => ({
-    top: getTop(line),
-    height: getSize(line),
-  }));
-  const isScrolledToEnd = useLastCallback(() => {
-    const last = getViewportLast();
-    return last === 1 || last === total;
-  });
-  const scrollToEnd = () => {
-    if (containerRef.current) {
-      containerRef.current.scrollTo(0, containerRef.current?.scrollHeight);
-    }
-  };
-  const getVisualLinesCount = useLastCallback(() => total);
-
-  useImperativeHandle(
-    ref,
-    () => ({
-      get container() {
-        return containerRef.current;
-      },
-      isScrolledToStart() {
-        return containerRef.current?.scrollTop === 0;
-      },
-      scrollToStart: () => {
-        if (containerRef.current) {
-          containerRef.current.scrollTo(0, 0);
+export const Console = forwardRef<ConsoleRef, ConsoleProps>(
```

Comment:
`isRunning` could be handled outside of the`Console` - it shouldn't disallow selecting the line when `!isRunning`, the external handler should decide what to do with the selected line

Reply:
We still need the `isRunning` boolean for scrolling to line number on render, as when the exeucution is ongoing we shouldn't scroll to the found query param

Reply:
Update: after latest commit, `isRunning` was removed and not used